src/lib/y2storage/device.rb

```diff
@@ -0,0 +1,98 @@
+require "y2storage/storage_class_wrapper"
+
+module Y2Storage
+  # An abstract base class of storage devices and a vertex in the Devicegraph.
+  #
+  # The Device class does not have a device name since some device types do
+  # not have a intrinsic device name, e.g. btrfs.
+  #
+  # This is a wrapper for Storage::Device
+  class Device
+    include StorageClassWrapper
+    wrap_class Storage::Device,
+      downcast_to: ["BlkDevice", "Mountable", "PartitionTables::Base"]
+
+    storage_forward :==
+    storage_forward :!=
+    storage_forward :sid
+    storage_forward :ancestors, as: "Device"
+    storage_forward :exists_in_devicegraph?
+    storage_forward :exists_in_probed?
+    storage_forward :exists_in_staging?
+    storage_forward :displayname
+    storage_forward :detect_resize_info
+    storage_forward :remove_descendants
+    storage_forward :userdata
+    storage_forward :userdata=
+
+    # Checks where the device is a disk
+    #
+    # @return [Boolean] true if this is a Y2Storage::Disk
+    def disk?
```

Comment:
to be honest this from my POV goes against idea of polymorphysm.
for you examples, I prefer to have something like
```
a = filesystem.disks
b = another_filesystem.ancestors.disks_or_partitions
```

I just feel a bit wrong to so heavily depends on kind of object. So if I transfer to common OOP example with animal these examples for me is like:

1. give me all animals, I do not mind its kind and then ask each animal if it is dog

2. give me all animals, I do not mind its kind and then look if it is dog

3. give me all dogs


I see this kind of methods also in past, but it is a bit different implemented. in base it was set to false and then in specialization to true.

What I see wrong in this approach? It is hard to extend. If new object is introduced, that act as BlkDevice, it cannot modify this class easily to be same of other kids.

Another part is that it adds to each child amount of public methods. This make introspection hard ( try now Disk.public__instance_methods.size ) what it returns. And public methods correspond for me also with responsibility. If we go back to animals example. You require that all animals can answers to questions like:
```
# dog
Are you cat? No.
Are you horse? No.
Are you dog? Yes.
Are you bird? No.
```
so for me it is also question about responsibilities. Also it breaks for me principle of "Tell, not ask" https://martinfowler.com/bliki/TellDontAsk.html . my approach is to tell, give me dogs. your approach for me looks like ask if you are dog and then act.

I hope I do not confuse you. Ask if something is not clear.